include/fiction/algorithms/simulation/sidb/assess_physical_population_stability.hpp

```diff
@@ -0,0 +1,299 @@
+                                                         params.physical_parameters.mu_plus());
+                                            critical_cell = c;
+                                            transition    = transition_type::NEUTRAL_TO_POSITIVE;
+                                        }
+                                    }
+                                    break;
+                                }
+                                case sidb_charge_state::POSITIVE:
+                                {
+                                    if (std::abs(-*charge_lyt.get_local_potential(c) +
+                                                 params.physical_parameters.mu_plus()) <
+                                        minimum_potential_difference_to_transition)
+                                    {
+                                        minimum_potential_difference_to_transition = std::abs(
+                                            -*charge_lyt.get_local_potential(c) + params.physical_parameters.mu_plus());
+                                        critical_cell = c;
+                                        transition    = transition_type::POSITIVE_TO_NEUTRAL;
+                                    }
+                                    break;
+                                }
+                                case sidb_charge_state::NONE:
+                                {
+                                    break;
+                                }
+                            }
+                        });
+                    valid_states_and_detailed_popstability_information.push_back(population_stability_information<Lyt>{
+                        critical_cell, transition, minimum_potential_difference_to_transition,
+                        convert_potential_to_distance(params.physical_parameters,
+                                                      minimum_potential_difference_to_transition,
+                                                      params.precision_for_distance_corresponding_to_potential),
+                        charge_lyt.get_system_energy()});
+                    break;
+                }
+            }
+        }
+
+        return valid_states_and_detailed_popstability_information;
+    };
+
+    /**
+     * Collects the system energy with the corresponding charge index information of all physically valid
+     * charge distributions of a given SiDB layout.
+     *
+     * @param sim_results The simulation results, including all physically valid charge distributions.
+     * @return A vector of energy_and_charge_index pairs, where each pair consists of a double value representing
+     * the system energy and a uint64_t representing the unique charge index. The vector is sorted in ascending order
+     * of the energy values.
+     */
+    [[nodiscard]] std::vector<energy_and_charge_index>
+    collect_energy_and_charge_index(const sidb_simulation_result<Lyt>& sim_results) noexcept
```

Comment:
Shouldn't this function be `private`?

Reply:
You are right. I just realized it too. Thank you!